persist-cli-tests/src/test/java/io/ballerina/persist/tools/utils/GeneratedSourcesTestUtils.java

```diff
@@ -112,6 +114,71 @@ public static void assertGeneratedSources(String subDir) {
         }
     }
 
+    public static void assertMigrateGeneratedSources(String subDir) {
+        Path generatedSources = Paths.get(GENERATED_SOURCES_DIRECTORY).resolve(subDir);
+        Path expectedSources = RESOURCES_EXPECTED_OUTPUT.resolve(subDir);
+
+        // check if directory content equals except the migrations directory
+        Assert.assertTrue(directoryContentEquals(expectedSources, generatedSources));
+
+        for (Path actualOutputFile: listFiles(generatedSources)) {
+            Path generatedMigrations = generatedSources.resolve(PERSIST_MIGRATIONS_DIR);
+            Path expectedMigrations = expectedSources.resolve(PERSIST_MIGRATIONS_DIR);
+            Assert.assertEquals(getNumOfSubDirectories(generatedMigrations),
+                    getNumOfSubDirectories(expectedMigrations));
+            try (Stream<Path> generatedFileList = Files.list(generatedMigrations)) {
+                generatedFileList.forEach(generatedDirectory -> {
+                    String generatedDirectoryNameWithoutTimeStamp =
+                            getDirectoryNameWithoutTimeStamp(generatedDirectory);
+                    assert(generatedDirectoryNameWithoutTimeStamp != null);
+                    try (Stream<Path> expectedFileList = Files.list(expectedMigrations)) {
+                        for (Path expectedDirectory : expectedFileList.toList()) {
+                            String expectedDirectoryNameWithoutTimeStamp =
+                                    getDirectoryNameWithoutTimeStamp(expectedDirectory);
+                            if (generatedDirectoryNameWithoutTimeStamp.equals(expectedDirectoryNameWithoutTimeStamp)) {
+                                Assert.assertTrue(directoryContentEquals(expectedDirectory, generatedDirectory));
+                                for (Path expectedFile : listFiles(expectedDirectory)) {
+                                    Path generatedFile = generatedDirectory.resolve(expectedFile.getFileName());
+                                    Assert.assertTrue(Files.exists(generatedFile));
+                                    Assert.assertEquals(readContent(generatedFile), readContent(expectedFile));
+                                }
+                                break;
+                            }
+                        }
+                    } catch (IOException e) {
+                        errStream.println(e.getMessage());
+                        Assert.fail();
+                    }
+                });
+            } catch (IOException e) {
+                errStream.println(e.getMessage());
+                Assert.fail();
+            }
+            errStream.println(actualOutputFile);
+        }
+    }
+
+    private static String getDirectoryNameWithoutTimeStamp(Path generatedDirectory) {
+        Path directory = generatedDirectory.getFileName();
+        if (Objects.isNull(directory)) {
+            Assert.fail();
```

Comment:
Shall we add an error message for `Assert.fail();`? In that case, it would be easier to pinpoint which particular assertion failed.

Reply:
Same applies to L150, L155 and L168